chemreg/jsonapi/filters.py

```diff
@@ -0,0 +1,22 @@
+import re
+
+from rest_framework_json_api import filters
+
+
+class QueryParameterValidationFilter(filters.QueryParameterValidationFilter):
+    """Allows setting additional query parameters in the view."""
+
+    view = None
+
+    @property
+    def query_regex(self):
+        query_params = ["sort", "include"]
+        if self.view:
+            query_params += getattr(self.view, "query_params", [])
```

Comment:
this is pulling `overrides`, plural, from the view, don't we want it to be `override`? Also, this seems to be getting called in a `list` view, but not in a `create`? Shouldn't it filter params for every view? Can we add the `query_params` attr on the viewset to only be added to the defaults on the `create` or `update` views?

Reply:
> this is pulling `overrides`, plural, from the view, don't we want it to be `override`?

Yes, absolutely. Thanks for catching that. I'll push up a fix.

> Also, this seems to be getting called in a `list` view, but not in a `create`? Shouldn't it filter params for every view? Can we add the `query_params` attr on the viewset to only be added to the defaults on the `create` or `update` views?

This is certainly possible. I think it would be a good idea. How it's implemented now is similar to how it's (erroneously) implemented in DRF JSON:API. For example, you could pass a `?filter` query parameter on a POST even though that wouldn't do anything.

Reply:
in reference to the `QueryParameterValidationFilter`, it seems like the backend filters are really setup to filter the queryset, not necessarily the params, there is  a `filter_queryset` method on the GenericAPIView in DRF, this gets called in the `get_object` method from this view. It might be possible to call `filter_queryset` in an overridden `create` method on the viewset? If these filter backends are setup for filtering the queryset it makes sense that it wouldn't apply to the `create` view.  Here's one way to get the validation to apply to a POST view as we would want:
```
    def create(self, request, *args, **kwargs):
        self.filter_queryset(self.get_queryset())
        return super().create(request, *args, **kwargs)
```

Reply:
Marking as resolved due to df10ac990afeefa53f6d4a8b24a844327950680e.